lib/plugins/aws/deployFunction/index.js

```diff
@@ -66,9 +66,15 @@ class AwsDeployFunction {
   }
 
   deployFunction() {
-    const artifactFileName = this.provider.naming
-      .getFunctionArtifactName(this.options.function);
-    const artifactFilePath = path.join(this.packagePath, artifactFileName);
+    const functionObject = this.serverless.service.getFunction(this.options.function);
+    const funcPackageConfig = functionObject.package || {};
+    let artifactFilePath = funcPackageConfig.artifact;
```

Comment:
@pmuens  I need help here. Whenever the `artifact` is read, I get some parsing errors and I'm not sure why that's happening. The error messages don't seem to have anything related to this part of the code. Can you run and have a look?